src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -24,6 +27,50 @@ public String getTotalPrice() {
         /**
          * Cakculates total price of shopping cart
          */
-        throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
+        //throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
+        int totalPrice = 0;
+        for (Product product: products) {
+            totalPrice += product.getPrice();
+        }
+        return Integer.toString(totalPrice);
+    }
+
+    public void addProduct(Product product) {
+        products.add(product);
+    }
+
+    public void removeProductByName(String laptop) {
```

Comment:
Not a good practice to have a specific item name as a method variable name. Try to go for something more generic like `product`